pkg/network/render.go

```diff
@@ -385,6 +388,46 @@ func isNetworkChangeSafe(prev, next *operv1.NetworkSpec, infraRes *bootstrap.Inf
 	return nil
 }
 
+func validateCidrExpansion(prev, next []operv1.ClusterNetworkEntry) bool {
+
+	// quick check to make sure clusterNetwork slices are of same size as we do not
+	// support adding/removing additional clusterNetwork entries
+	if len(prev) != len(next) {
+		log.Printf("Adding/Removing clusterNetwork entries of the same type is not supported")
+		return false
+	}
+
+	// CIDR Mask expansion is only supported for OVNK.
+
+	// since we do not allow the clusterNetwork[] size to change, it should be safe to compare
```

Comment:
I am not sure if that is a safe assumption, users could potentially reorder `clusterNetwork`?

Reply:
good catch. next update will sort prev and next each time to make it deterministic